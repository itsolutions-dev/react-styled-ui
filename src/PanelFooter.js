import styled, { css } from './styledComponent';
import {
  getSizeOrDefault,
  getBackgroundColor,
} from './utils/';

const PanelFooter = styled({
  component: 'div',
  style: css`
    font-size: 20px;
    font-weight: 500;
    padding: 10px 20px;
    color: ${props => getBackgroundColor(props)};
    ${(props) => {
      const color = props.borderColor !== undefined ? props.borderColor : getBackgroundColor(props);
      return `
        border-top: 1px solid ${color};
        border-left: 1px solid ${color};
        border-bottom: 1px solid ${color};
        border-right: 1px solid ${color};
      `;
    }}
    border-bottom-left-radius: ${props => getSizeOrDefault(props.radius, props.theme.radius)};
    border-bottom-right-radius: ${props => getSizeOrDefault(props.radius, props.theme.radius)};
    border-radius: 0 !important;
  `,
});

PanelFooter.defaultProps = {
  primary: true,
};

export default PanelFooter;
