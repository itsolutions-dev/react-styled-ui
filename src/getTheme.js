import defaultTheme from './theme';
import deepMerge from './utils/deepMerge';

const getTheme = deepMerge.bind(null, defaultTheme);

export default getTheme;
