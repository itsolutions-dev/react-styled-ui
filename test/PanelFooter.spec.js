import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import getTheme from '../src/getTheme';
import PanelFooter from '../src/PanelFooter';

const theme = getTheme({
  radius: '8px',
  palette: {
    primary: 'blue',
  },
});

describe('PanelFooter', () => {
  test('should render a PanelFooter', () => {
    const component = ReactTestRenderer.create(<PanelFooter theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<PanelFooter theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have the given border color', () => {
    const component = ReactTestRenderer.create(<PanelFooter borderColor="green" theme={theme} />);
    expect(component).toHaveStyleRule('border-top', '1px solid green');
    expect(component).toHaveStyleRule('border-bottom', '1px solid green');
    expect(component).toHaveStyleRule('border-left', '1px solid green');
    expect(component).toHaveStyleRule('border-right', '1px solid green');
  });

  test('should be primary by default', () => {
    const component = ReactTestRenderer.create(<PanelFooter theme={theme} />);
    expect(component).toHaveStyleRule('color', 'blue');
    expect(component).toHaveStyleRule('border-top', '1px solid blue');
    expect(component).toHaveStyleRule('border-bottom', '1px solid blue');
    expect(component).toHaveStyleRule('border-left', '1px solid blue');
    expect(component).toHaveStyleRule('border-right', '1px solid blue');
  });

  test('should have the given radius', () => {
    const component = ReactTestRenderer.create(<PanelFooter radius={10} theme={theme} />);
    expect(component).toHaveStyleRule('border-bottom-left-radius', '10px');
    expect(component).toHaveStyleRule('border-bottom-right-radius', '10px');
  });

  test('should get radius from theme', () => {
    const component = ReactTestRenderer.create(<PanelFooter theme={theme} />);
    expect(component).toHaveStyleRule('border-bottom-left-radius', '8px');
    expect(component).toHaveStyleRule('border-bottom-right-radius', '8px');
  });
});
