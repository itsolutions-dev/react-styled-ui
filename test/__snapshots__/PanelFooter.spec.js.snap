// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelFooter should render a PanelFooter 1`] = `
.c0 {
  font-size: 20px;
  font-weight: 500;
  padding: 10px 20px;
  color: blue;
  border-top: 1px solid blue;
  border-left: 1px solid blue;
  border-bottom: 1px solid blue;
  border-right: 1px solid blue;
  border-bottom-left-radius: 8px;
  border-bottom-right-radius: 8px;
  border-radius: 0 !important;
}

<div
  className="c0"
/>
`;
